multi_line_string.go

```diff
@@ -1,7 +1,23 @@
 package geom
 
+// MultiLineString is a geometry with multiple LineStrings.
 type MultiLineString [][][2]float64
 
-func (mls MultiLineString) LineStrings() [][][2]float64 {
-	return mls
+// LineStrings returns the coordinates for the linestrings
+func (mls *MultiLineString) LineStrings() [][][2]float64 {
+	return *mls
+}
+
+// Points returns a slice of XY values
+func (mls *MultiLineString) Points() (points [][2]float64) {
+	for _, ls := range *mls {
+		points = append(points, ls...)
+	}
+	return
+}
+
+// SetLineStrings modifies the array of 2D coordinates
+func (mls *MultiLineString) SetLineStrings(input [][][2]float64) (err error) {
+	*mls = append((*mls)[:0], input...)
```

Comment:
Is it idiomatic to check for null in every pointer receiver? It seems to me that this would mask a failure condition that needs to be caught in the tests.

Or did you mean the input?

Reply:
No mls can be nil. Nil is a valid value for a pointer.

Reply:
Sure nil is a valid value for a pointer, but whether a code segfaults has nothing to do with whether pointer receivers are used and everything to do with trying to dereference a nil pointer. Calling a method with a non-pointer receiver on a nil object will still segfault when you dereference it. Same with calling a method on a nil interface (the expected way to use this library). Your recommendation would just add boilerplate to every pointer receiver without ever improving application safety.

Reply:
I will not let you call a method with a non-pointer receiver on a nil object. You don't dereference a non-pointer receiver. 

If your receiver is a pointer then you should be checking if it's nil. As nil is a valid value for that receiver, and if you don't feel it is, you should return an error saying so. Panicking is not appropriate. Panic should be used for unrecoverable errors. Passing in nil, not a unrecoverable error.